src/bot.py

```diff
@@ -273,10 +273,14 @@ def on_maintenance(self):
 
         # Check if we are out of tps or need repairing
         need_repair = self._ui_manager.repair_needed()
-        if self._tps_left < random.randint(3, 5) or need_repair or self._config.char["always_repair"]:
+        if self._tps_left < random.randint(3, 5) or need_repair or self._config.char["always_repair"] or self._config.char["always_refill_keys"] & self._config.routes["run_arcane"]:
             if need_repair: Logger.info("Repair needed. Gear is about to break")
             else: Logger.info("Repairing and buying TPs at next Vendor")
-            self._curr_loc = self._town_manager.repair_and_fill_tps(self._curr_loc)
+            if self._config.char["always_refill_keys"] & self._config.routes["run_arcane"]:
```

Comment:
keep code dry. you use twice the same condition (always_refill_keys and run_arcane). Store it in a variable similar to how it's done with need_repair. Something like `buy_keys`